### PR TITLE
feat(proto): add per-rule ansible-core version applicability (ADR-057)

### PR DESCRIPTION
**Pull request by cidrblock**
## Summary
- M rules fire unconditionally regardless of the target ansible-core version, so users scanning for 2.19 see violations about 2.23/2.24 changes with no version context. This PR adds PEP 440 version specifiers to each rule's metadata so the UI can filter violations by version relevance.

## Changes
- **ADR-057**: new architectural decision record documenting the design (PEP 440 specifiers, central table, UI-side filtering)
- **Proto**: add `ansible_core_version` string field to `RuleDefinition` in `reporting.proto`
- **`version_defaults.py`**: new central rule → `SpecifierSet` mapping for all M-series rules (implemented only), following the `severity_defaults.py` pattern (ADR-043)
- **`rule_catalog.py`**: populate `ansible_core_version` on `RuleDefinition` during catalog registration
- **`violation_convert.py`**: stamp `ansible_core_version` into `Violation.metadata` map on dict→proto conversion; preserve explicit values from input dict (plugin/round-trip safe); central table as fallback; empty string treated as absent
- **Frontmatter**: add `ansible_core_version` to all 27 M-rule `.md` sidecar files (native, OPA, ansible validators)
- **Tests**: 22 tests covering `version_defaults`, `rule_catalog`, and `violation_convert` changes — including frontmatter↔table consistency checks, edge cases for empty/malformed version strings, explicit value preservation, plugin rule round-trips, and empty-string fallback

## Test plan
- [x] `tox -e lint` passes (ruff, mypy, pydoclint, adr-index, rule-catalog all green)
- [x] `tox -e unit` passes (2676 passed, 61.48% coverage)
- [x] ADR-057 added
- [x] Rule catalog auto-regenerated
- [x] Cold subagent review findings addressed (4/4 fixed)
- [x] Copilot review findings addressed (3/3 fixed — round-trip fidelity, empty-string edge case)
- [x] Frontmatter ↔ VERSION_DEFAULTS consistency tests added

### CODE REVIEW

File: src/apme_engine/daemon/violation_convert.py [hunk 1]
```diff
@@ -220,6 +221,10 @@ def violation_dict_to_proto(v: ViolationDict | Mapping[str, str | int | list[int
             else:
                 out.metadata[key] = str(val)
 
+    version_spec = get_version_spec_str(out.rule_id)
+    if version_spec:
+        out.metadata["ansible_core_version"] = version_spec
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/apme_engine/daemon/violation_convert.py]: Added `ansible_core_version` to `_METADATA_KEYS` so it flows in from the input dict, and changed the central-table injection to a fallback (`if "ansible_core_version" not in out.metadata`). Plugin rules and round-trips now preserve explicit values. Added tests for both scenarios. Fixed in 2bd1c4f.

File: src/apme_engine/daemon/violation_convert.py [hunk 2]
```diff
@@ -220,6 +221,10 @@ def violation_dict_to_proto(v: ViolationDict | Mapping[str, str | int | list[int
...
+
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/apme_engine/daemon/violation_convert.py]: Same fix as above — `ansible_core_version` is now in `_METADATA_KEYS` for dict→proto flow, and the central table only fills it as a fallback. Also removed the dead `elif` branch in `violation_proto_to_dict` since `_METADATA_KEYS` membership handles it. Fixed in 2bd1c4f.

File: src/apme_engine/daemon/violation_convert.py [hunk 3]
```diff
@@ -220,6 +222,11 @@ def violation_dict_to_proto(v: ViolationDict | Mapping[str, str | int | list[int
             else:
                 out.metadata[key] = str(val)
 
+    if "ansible_core_version" not in out.metadata:
+        version_spec = get_version_spec_str(out.rule_id)
+        if version_spec:
+            out.metadata["ansible_core_version"] = version_spec
+
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply src/apme_engine/daemon/violation_convert.py]: Changed the fallback guard from `"ansible_core_version" not in out.metadata` to `not out.metadata.get("ansible_core_version")`, so both absent and empty-string values fall through to the VERSION_DEFAULTS table for known M-rules. Added a test for the empty-string edge case. Fixed in b0daa81.
